amount.go

```diff
@@ -373,3 +369,21 @@ func decimalContext(decimals ...*apd.Decimal) *apd.Context {
 	}
 	return decimalContextPrecision19
 }
+
+// decimalContextWithRoundingMode returns the decimal context to use for a calculation with the given round mode.
+// It optimizes for the most common RoundHalfUp mode by returning a preallocated global context for it.
+func decimalContextWithRoundingMode(mode RoundingMode, decimals ...*apd.Decimal) *apd.Context {
```

Comment:
Let's call this func roundingContext(), since that's what it's used for. And since we know we only round a single number, we can reduce the number of decimal parameters.

For example:
```
// roundingContext returns the decimal context to use for rounding.
func roundingContext(decimal *apd.Decimal, mode RoundingMode) *apd.Context {
```